stable/database/files/nuosm

```diff
@@ -576,12 +605,20 @@ if [ "$myArchive" -ne "-1" ]; then
       # restore chart doesn't have notion of stream|backupset
       # `nuodocker restore archive` works with both
       restore_type="backupset"
+      if [ "$restore_source" == ":snapshot" ]; then
+        # the archive content has been restored in-place from a snapshot by an
+        # external backup manager (not by NuoDB Helm charts or NuoDB operator);
+        # we will get here only if the Kubernetes Aware Admin (KAA) is disabled
+        # since otherwise the archive metadata will be removed by KAA when the
+        # PVC is re-created with a snapshot reference
+        restore_type="snapshot"
```

Comment:
If KAA is enabled, is this unnecessary? Or does KAA have to be disabled?

Reply:
KAA has to be disabled. Otherwise, the old archives (those for which restore is requested) are purged when BOaaS re-creates the PVCs.

Reply:
The database restore request is the KV store, so it should be straightforward to check it. I will make a change.